src/include/storage/index/index.h

```diff
@@ -169,6 +186,35 @@ class Index {
    * @return IndexKeyKind selected by the IndexBuilder at index construction
    */
   IndexKeyKind KeyKind() const { return metadata_.KeyKind(); }
+
+  /**
+   * Waits until the is_live_ variable is true
+   */
+  void WaitUntilLive() const {
+    std::unique_lock<std::mutex> lock(is_live_mutex_);
+    while (!is_live_) {
+      is_live_condvar_.wait(lock);
+    }
+    lock.unlock();
+  }
+
+  /**
+   * Sets the is_live_ variable to true and notifies any waiters on it
+   */
+  void SetLive() {
+    {
+      std::lock_guard<std::mutex> lock(is_live_mutex_);
+      is_live_ = true;
+    }
+    is_live_condvar_.notify_all();
+  }
+
+  /**
+   * Sets the index as not being live yet. Used while populating
+   */
+  void SetNotLive() {
```

Comment:
why doesn't this require locking?

Reply:
The name `SetNotLive` is awkward. Why not just make it `SetIsLive(bool)`?